compiler/compile.bal

```diff
@@ -149,3 +154,21 @@ function subModuleSourceParts(string basename, bir:ModuleId id) returns front:So
 function subModuleSuffix(bir:ModuleId id) returns string {
     return ".".'join(...id.names.slice(1));
 }
+
+function buildInitModule(CompileContext cx, front:ResolvedModule entryMod) returns CompileError? {
```

Comment:
We shouldn't use the same name for this as for the method on CompileContext when they do different things. buildModule is returning a llvm:MOdule, so I would call this something different, perhaps generateInitModule.